internal/rbd/controllerserver.go

```diff
@@ -76,12 +76,13 @@ func (cs *ControllerServer) validateVolumeReq(ctx context.Context, req *csi.Crea
 	}
 	options := req.GetParameters()
 	if value, ok := options["clusterID"]; !ok || value == "" {
-		return status.Error(codes.InvalidArgument, "missing or empty cluster ID to provision volume from")
+		return status.Error(codes.InvalidArgument, "empty cluster ID to provision volume from")
 	}
 	if value, ok := options["pool"]; !ok || value == "" {
-		return status.Error(codes.InvalidArgument, "missing or empty pool name to provision volume from")
+		if value, ok := options["topologyConstrainedPools"]; !ok || value == "" {
+			return status.Error(codes.InvalidArgument, "empty pool name or topologyConstrainedPools to provision volume")
+		}
 	}
```

Comment:
```suggestion
poolValue, poolOK := options["pool"]
topologyConstrainedPoolsValue, topologyOK := options["topologyConstrainedPools"]

if !poolOK {
    if topologyOK && topologyConstrainedPoolsValue == "" {
        return status.Error(codes.InvalidArgument, "empty pool name or topologyConstrainedPools to provision volume")
    } else {
        return status.Error(codes.InvalidArgument, "missing or empty pool name to provision volume from")
    }
} else if poolValue == "" {
    return status.Error(codes.InvalidArgument, "missing or empty pool name to provision volume from")
}
```

Can you refractor like above to consider all the cases
* For non topology case
  * when key is not set
  * when key is set but value is empty 
* For non topology case
  * when the key is not set
  * when the key is set but value is empty 

Make the adjustment as required to cover above cases

Reply:
updated,

Reply:
this would be 
>else {
        return status.Error(codes.InvalidArgument, "missing or empty pool name to provision volume from")
    }
    
    ----> 
    } else if !topologyOK {
			return status.Error(codes.InvalidArgument, "missing or empty pool name to provision volume from")
		}